applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_generic_utilities.cpp

```diff
@@ -56,4 +58,35 @@ KRATOS_TEST_CASE_IN_SUITE(CheckMatrixPermutation, KratosGeoMechanicsFastSuiteWit
     KRATOS_EXPECT_MATRIX_EQ(GenericUtilities::MatrixWithPermutedColumns(matrix, indices), expected_result);
 }
 
+KRATOS_TEST_CASE_IN_SUITE(CollectIdsFromEntity_ReturnsEmptyListForEmptyGeometry, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto node_ids = GenericUtilities::CollectIdsFromEntity(Geometry<Node>{});
+
+    KRATOS_EXPECT_TRUE(node_ids.empty())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CollectIdsFromEntity_ReturnsCorrectNodeIds, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arange
+    PointerVector<Node> nodes;
+    nodes.push_back(Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(3, 0.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(42, 0.0, 0.0, 0.0));
+    nodes.push_back(Kratos::make_intrusive<Node>(314, 0.0, 0.0, 0.0));
+    const Geometry geometry(1, nodes);
+
+    // Act
+    const auto node_ids_vector = GenericUtilities::CollectIdsFromEntity(geometry);
+
+    std::set<Node::IndexType> node_ids_set;
+    GenericUtilities::CollectIdsFromEntity(geometry, node_ids_set);
```

Comment:
The same holds here:

```suggestion
    GenericUtilities::CollectIdsFromEntity(geometry, std::inserter(node_ids_set, node_ids_set.end()));
```